Common/Utils/include/CommonUtils/BoostHistogramUtils.h

```diff
@@ -0,0 +1,243 @@
+// or submit itself to any jurisdiction.
+
+/// \file   BoostHistogramUtils.h
+/// \author Hannah Bossi, hannah.bossi@yale.edu
+
+#include <cmath>
+#include <numeric>
+#include <algorithm>
+#include <vector>
+#include <array>
+
+#include "Rtypes.h"
+#include "TLinearFitter.h"
+#include "TVectorD.h"
+#include "TMath.h"
+#include "TF1.h"
+#include "Foption.h"
+#include "HFitInterface.h"
+#include "TFitResultPtr.h"
+#include "TFitResult.h"
+#include "Fit/Fitter.h"
+#include "Fit/BinData.h"
+#include "Math/WrappedMultiTF1.h"
+
+namespace o2
+{
+namespace utils
+{
+
+template<typename AxisIterator>
+class BinCenterView {
+    public:
+        BinCenterView(AxisIterator iter)
+        AxisIterator& operator++() {
+            ++mBaseIterator;
+            return mBaseIterator;
+        }
+        AxisIterator operator++(int) {
+            AxisIterator result(mBaseIterator);
+            mBaseIterator++;
+            return result;
+        }
+
+        decltype(auto) operator*() {return mBaseIterator->center(); }
+    private:
+        AxisIterator    mBaseIterator;
+};
+
+template<typename AxisIterator>
+class BinUpperView {
+    public:
+        BinCenterView(AxisIterator iter)
+        AxisIterator& operator++() {
+            ++mBaseIterator;
+            return mBaseIterator;
+        }
+        AxisIterator operator++(int) {
+            AxisIterator result(mBaseIterator);
+            mBaseIterator++;
+            return result;
+        }
+
+        decltype(auto) operator*() {return mBaseIterator->upper(); }
+    private:
+        AxisIterator    mBaseIterator;
+};
+
+template<typename AxisIterator>
+class BinLowerView {
+    public:
+        BinCenterView(AxisIterator iter)
+        AxisIterator& operator++() {
+            ++mBaseIterator;
+            return mBaseIterator;
+        }
+        AxisIterator operator++(int) {
+            AxisIterator result(mBaseIterator);
+            mBaseIterator++;
+            return result;
+        }
+
+        decltype(auto) operator*() {return mBaseIterator->lower(); }
+    private:
+        AxisIterator    mBaseIterator;
+};
+
+/// fast fit of an array with ranges (histogram) with gaussian function
+///
+/// Fitting procedure:
+/// 1. Step - make logarithm
+/// 2. Linear  fit (parabola) - more robust, always converges, fast
+///
+/// \return chi2 or exit code
+///          >0: the chi2 returned by TLinearFitter
+///          -3: only three points have been used for the calculation - no fitter was used
+///          -2: only two points have been used for the calculation - center of gravity was uesed for calculation
+///          -1: only one point has been used for the calculation - center of gravity was uesed for calculation
+///          -4: invalid result!!
+///
+template <typename Iterator, typename AxisIterator, typename T>
+double fitGaus(Iterator first, Iterator last, AxisIterator axisfirst,  std::vector<T>& param)
```

Comment:
I suggest to move the function to MathUtils and create a WrapperFunction taking the boost histogram directly. Also I would recommend to create a small struct `FitResult` containing params and chi2 instead of returning params as reference.